File: src/main/java/com/xebia/finance/contract/service/ContractService.java
package com.xebia.finance.contract.service;

import com.xebia.finance.contract.dto.ContractResponse;
import com.xebia.finance.contract.dto.CreateContractRequest;
import com.xebia.finance.contract.dto.UpdateContractRequest;

import java.util.List;
import java.util.UUID;

/**
 * Service interface for managing contracts.
 *
 * @author Pratik Kumar | Xebia Virtual Internship | 70812345pratik@gmail.com
 */
public interface ContractService {

    /**
     * Creates a new contract.
     */
    ContractResponse createContract(CreateContractRequest request);

    ContractResponse updateContract(UUID id, UpdateContractRequest request);

    /**
     * Retrieves all contracts.
     */
    List<ContractResponse> getAllContracts();

    /**
     * Retrieves a contract by ID.
     */
    ContractResponse getContractById(UUID id);

    /**
     * Deletes a contract.
     */
    void deleteContract(UUID id);
}

File: finance-ui/src/services/contractService.js
import api from "./api";

export const getContracts = () => api.get("/contracts");

export const createContract = (contract) =>
  api.post("/contracts", contract);

export const updateContract = (id, contract) =>
  api.put(`/contracts/${id}`, contract);

export const deleteContract = (id) =>
  api.delete(`/contracts/${id}`);